### DISCUSSION TITLE
Fixture.Create fails for some common types saying they have a circular reference

### DISCUSSION THREAD

**Discussion by jamesvickers19** [AutoFixture/AutoFixture#1229] [Q&A] [answered]
Was trying to use AutoFixture to create a couple of basic types but it appears unable to without extra setup which is a bummer.  For example this doesn't work:

```csharp
new Fixture().Create<System.Drawing.Rectangle>();
```

Same for `System.Drawing.Point`, `System.Drawing.Size`.  All of these examples throw:

```
AutoFixture.ObjectCreationExceptionWithPath : AutoFixture was unable to create an instance of type AutoFixture.Kernel.SeededRequest because the traversed object graph contains a circular reference . . .
```

For `Rectangle` for instance it says the circular reference is:

```
Request path:
    	System.Drawing.Point pt
    	  System.Drawing.Size
    	    System.Drawing.Size sz
    	      System.Drawing.Point
    	        System.Drawing.Point location
    	          System.Drawing.Rectangle
```

I don't think these types have circular references but I could easily be mistaken.  Is there some way to make AutoFixture create them out-of-the-box?

Thanks in advance and really appreciate this library.

**Comment by ecampidoglio** [comment, marked as answer]
You can tell AutoFixture to stop recursing after a certain number of iterations. The exception message contains instructions on how to do that:

> ObjectCreationExceptionWithPath: AutoFixture was unable to create an instance of type AutoFixture.Kernel.SeededRequest because the traversed object graph contains a circular reference. Information about the circular path follows below. This is the correct behavior when a Fixture is equipped with a ThrowingRecursionBehavior, which is the default. This ensures that you are being made aware of circular references in your code. Your first reaction should be to redesign your API in order to get rid of all circular references. However, if this is not possible (most likely because parts or all of the API is delivered by a third party), you can replace this default behavior with a different behavior: on the Fixture instance, remove the ThrowingRecursionBehavior from Fixture.Behaviors, and instead add an instance of OmitOnRecursionBehavior:
>
> ```
> fixture.Behaviors.OfType<ThrowingRecursionBehavior>().ToList().ForEach(b => fixture.Behaviors.Remove(b));
> fixture.Behaviors.Add(new OmitOnRecursionBehavior());
> ```

You can specify the maximum depth of your object graph through the constructor of `OmitOnRecursionBehavior`. The [default value is `1`](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoFixture/OmitOnRecursionBehavior.cs#L12).

**Reply by jamesvickers19** [reply]
Thanks so much!

**Reply by ecampidoglio** [reply]
Glad I could help. 🙂

Could you please [mark the question as answered](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer)?
